src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -75,6 +75,13 @@ object ScalafixPlugin extends AutoPlugin {
         "Optional location to .scalafix.conf file to specify which scalafix rules should run. " +
           "Defaults to the build base directory if a .scalafix.conf file exists."
       )
+
+    val scalafixCallback: SettingKey[ScalafixMainCallback] =
+      settingKey[ScalafixMainCallback](
+        "The handler for the diagnostics emitted during scalafix execution. Can be set in ThisBuild or at project-level. " +
```

Comment:
Since you went for (2) in https://github.com/scalacenter/sbt-scalafix/pull/384#discussion_r1462919626
```suggestion
        "The handler for the diagnostics emitted during scalafix execution. Must be set in ThisBuild. " +
```